.github/workflows/build-fedora.sh

```diff
@@ -16,44 +21,111 @@ for phase in "${PHASES[@]}"; do
+            dnf -y install \
+                dnf-plugins-core \
+                dnsmasq \
+                iproute \
+                kernel-core \
+                mkosi \
+                python3-pyxattr \
+                qemu-kvm \
+                scsi-target-utils \
+                systemd \
+                zstd
             ;;
-        INITRD)
+        INITRD_BASIC)
             rm -fr mkosi.output
             mkdir mkosi.output
-
-            python3 -m mkosi --default fedora.mkosi \
+            # Build a basic initrd
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
+                             --default fedora.mkosi \
                              --image-version="$KVER" \
                              --environment=KERNEL_VERSION="$KVER" \
                              -f build
             # Check if the image was indeed generated
             stat "mkosi.output/initrd_$KVER.cpio.zstd"
 
-            # Basic boot test
+            # Build a basic OS image to test the initrd with
+            rm -fr _rootfs
+            mkdir _rootfs
+            pushd _rootfs
+            # shellcheck source=/dev/null
+            source <(grep -E "(ID|VERSION_ID)" /etc/os-release)
+            mkosi --cache "$MKOSI_CACHE" \
+                  --distribution="$ID" \
+                  --release="$VERSION_ID" \
+                  --format=gpt_btrfs \
+                  --output=rootfs.img
+            popd
+
+            # Sanity check if the initrd is bootable
             timeout -k 10 5m qemu-kvm -m 512 -smp "$(nproc)" -nographic \
                                       -initrd "mkosi.output/initrd_$KVER.cpio.zstd" \
                                       -kernel "/usr/lib/modules/$KVER/vmlinuz" \
                                       -append "rd.systemd.unit=systemd-poweroff.service rd.debug $SYSTEMD_LOG_OPTS console=ttyS0"
+
+            # Boot the initrd with an OS image
+            timeout -k 10 5m qemu-kvm -m 1024 -smp "$(nproc)" -nographic \
+                                      -initrd "mkosi.output/initrd_$KVER.cpio.zstd" \
+                                      -kernel "/usr/lib/modules/$KVER/vmlinuz" \
+                                      -drive "format=raw,cache=unsafe,file=_rootfs/rootfs.img" \
+                                      -append "root=LABEL=root rd.debug $SYSTEMD_LOG_OPTS console=ttyS0 systemd.unit=systemd-poweroff.service systemd.default_timeout_start_sec=240"
+
+            # Cleanup
+            rm -fr mkosi.output _rootfs
             ;;
-        INITRD_FULLBOOT)
-            if [[ ! -e "mkosi.output/initrd_$KVER.cpio.zstd" ]]; then
-                echo >&2 "Missing initrd image (run the INITRD phase first)"
-                exit 1
-            fi
+        INITRD_ISCSI)
+            rm -fr mkosi.output
+            mkdir mkosi.output
+            # Build the initrd with iSCSI support
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
```

Comment:
`python3 -m` ?

Reply:
Hah, force of habit I guess from local testing, replaced in the new tests, but also in the existing ones.